react/forms.rst

```diff
@@ -0,0 +1,206 @@
+                },
+                {
+                  question: "Why does Plone need a Foundation?",
+                  answer:
+                    "Plone has reached critical mass, with enterprise implementations and worldwide usage. The Foundation is able to speak for Plone, and provide strong and consistent advocacy for both the project and the community. The Plone Foundation also helps ensure a level playing field, to preserve what is good about Plone as new participants arrive."
+                }
+              ],
+              question: "",
+              answer: ""
+            };
+          }
+
+          onDelete(index) {
+            let faq = this.state.faq;
+            faq.splice(index, 1);
+            this.setState({
+              faq
+            });
+          }
+
+          onChangeQuestion(event) {
+            this.setState({
+              question: event.target.value
+            });
+          }
+
+          onChangeAnswer(event) {
+            this.setState({
+              answer: event.target.value
+            });
+          }
+
+          render() {
+            return (
+              <div>
+                <ul>
+                  {this.state.faq.map((item, index) => (
+                    <FaqItem
+                      question={item.question}
+                      answer={item.answer}
+                      index={index}
+                      onDelete={this.onDelete}
+                    />
+                  ))}
+                </ul>
+                <form>
+                  <label>
+                    Question:
+                    <input
+                      name="question"
+                      type="text"
+                      value={this.state.question}
+                      onChange={this.onChangeQuestion}
+                    />
+                  </label>
+                  <label>
+                    Answer:
+                    <textarea
+                      name="answer"
+                      value="this.state.answer"
+                      onChange={this.onChangeAnswer}
+                    />
+                  </label>
+                  <input type="submit" value="Add" />
+                </form>
+              </div>
+            );
+          }
+        }
+
+        export default App;
+
+
+Submit Handler
+==============
+
+Now that our values are managed in the state we can write our submit handler.
+Write an :file:`onSubmit` handler which reads the values from the state and
+add the new Faq item to the list. After the item is added the inputs should also
+reset to empty values.
+
+..  admonition:: Solution
+    :class: toggle
+
+    ::
```

Comment:
```
    .. code-block::
        :language: js
```

OK, you get the idea. Explicitly naming the language is better for syntax highlighting.